inc/knowbaseitemcategory.class.php

```diff
@@ -92,125 +90,141 @@ static function showFirstLevel($options) {
             return false;
          }
 
+         $where = [];
          if (Session::getLoginUserID()) {
-            $faq_limit = getEntitiesRestrictRequest("AND", "glpi_knowbaseitemcategories", "", "",
-                                                    true);
+            $where = getEntitiesRestrictCriteria('glpi_knowbaseitemcategories', '', '', true);
          } else {
             // Anonymous access
             if (Session::isMultiEntitiesMode()) {
-               $faq_limit = " AND (`glpi_knowbaseitemcategories`.`entities_id` = 0
-                                   AND `glpi_knowbaseitemcategories`.`is_recursive` = 1)";
+               $where['glpi_knowbaseitemcategories.entities_id'] = 0;
+               $where['glpi_knowbaseitemcategories.is_recursive'] = 1;
             }
          }
 
          // Get All FAQ categories
          if (!isset($_SESSION['glpi_faqcategories'])) {
-
-            $_SESSION['glpi_faqcategories'] = '(0)';
+            $_SESSION['glpi_faqcategories'] = [];
```

Comment:
I may be wrong, but the only place where `$_SESSION['glpi_faqcategories']` is read and write is on this `showFirstLevel` method, which is not used anymore and is deprecated.
We should remove this session var before publishing the 9.4.0 version.